packages/@uppy/core/src/Uppy.js

```diff
@@ -434,6 +434,18 @@ class Uppy {
    * The `files` value is passed in because it may be updated by the caller without updating the store.
    */
   #checkAndCreateFileStateObject (files, fileDescriptor) {
+    // Uppy expects files in { name, type, size, data } format.
+    // If the actual File object is passed from input[type=file] or drag-drop,
+    // we normalize it to match Uppy file object
+    if (!fileDescriptor.data && !fileDescriptor.isRemote) {
+      fileDescriptor = {
+        name: fileDescriptor?.name,
+        type: fileDescriptor?.type,
+        size: fileDescriptor?.size,
```

Comment:
If `fileDescriptor` is nullish, line 440 would have crashed, so we don't need optional chaining here.
```suggestion
        name: fileDescriptor.name,
        type: fileDescriptor.type,
        size: fileDescriptor.size,
```